tests/run-make/translation/rmake.rs

```diff
@@ -0,0 +1,178 @@
+//! Smoke test for the rustc diagnostics translation infrastructure.
+//!
+//! # References
+//!
+//! - Current tracking issue: <https://github.com/rust-lang/rust/issues/132181>.
+//! - Old tracking issue: <https://github.com/rust-lang/rust/issues/100717>
+//! - Initial translation infra implementation: <https://github.com/rust-lang/rust/pull/95512>.
+
+// This test uses symbolic links to stub out a fake sysroot to save testing time.
+//@ needs-symlink
+// Needs to spawn `rustc` subprocesses.
+// FIXME(#128295): clean these ignores up in favor of `needs-subprocess` directive
+//@ ignore-wasm
+//@ ignore-emscripten
+//@ ignore-sgx
```

Comment:
I really should implement this lol

Reply:
#135926, I'll send a follow-up after *that* merges, didn't want to block this PR on *that* PR

Reply:
Fixed